### PR TITLE
fix : App crash on comment click fixed

### LINKED ISSUE #2452: App crash on clicking Comments Section in Live Feed
**Actual Behaviour**

The app crashes on click of comments section in Live Feed. It occur only when there are some comments present.

**Expected Behaviour**

This should not occur.

**LogCat for the issue**

`java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.fossasia.openevent.core.feed.facebook.api.Commenter.getName()' on a null object reference
                                                                            at org.fossasia.openevent.core.feed.facebook.CommentsListAdapter.onBindViewHolder(CommentsListAdapter.java:77)
                                                                            at org.fossasia.openevent.core.feed.facebook.CommentsListAdapter.onBindViewHolder(CommentsListAdapter.java:22)
                                                                            at android.support.v7.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:6673)
                                                                            at android.support.v7.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:6714)
                                                                            at android.support.v7.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:5647)
                                                                            at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5913)
                                                                            at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5752)
                                                                            at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5748)
                                                                            at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2232)
                                                                            at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1559)
                                                                            at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1519)
                                                                            at android.support.v7.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:614)
                                                                            at android.support.v7.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:3812)
                                                                            at android.support.v7.widget.RecyclerView.onMeasure(RecyclerView.java:3225)
                                                                            at android.view.View.measure(View.java:18809)
                                                                            at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5954)
                                                                            at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1465)
                                                                            at android.widget.LinearLayout.measureVertical(LinearLayout.java:748)
                                                                            at android.widget.LinearLayout.onMeasure(LinearLayout.java:630)
                                                                            at android.view.View.measure(View.java:18809)
                                                                            at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5954)
                                                                            at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
                                                                            at android.view.View.measure(View.java:18809)
                                                                            at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5954)
                                                                            at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
                                                                            at android.view.View.measure(View.java:18809)
                                                                            at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5954)
                                                                            at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
                                                                            at com.android.internal.policy.PhoneWindow$DecorView.onMeasure(PhoneWindow.java:2643)
                                                                            at android.view.View.measure(View.java:18809)
                                                                            at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:2112)
                                                                            at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1203)
                                                                            at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1464)
                                                                            at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1119)
                                                                            at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6060)
                                                                            at android.view.Choreographer$CallbackRecord.run(Choreographer.java:858)
                                                                            at android.view.Choreographer.doCallbacks(Choreographer.java:670)
                                                                            at android.view.Choreographer.doFrame(Choreographer.java:606)
                                                                            at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
                                                                            at android.os.Handler.handleCallback(Handler.java:746)
                                                                            at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                            at android.os.Looper.loop(Looper.java:148)
                                                                            at android.app.ActivityThread.main(ActivityThread.java:5443)
                                                                            at java.lang.reflect.Method.invoke(Native Method)
                                                                            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728)
                                                                            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
05-14 04:53:57.410 17848-17918/? E/PlaceStateUpdater: Received no places`

**Screenshots of the issue**

videotogif_2018 05 14_05 01 23

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by dreadpool2**
Fixes #2452 

It seems that the JSON response from the graph API returns no user name of the person who commented and it only returns the "from" object if FOSSASIA comments. We can't access the names of the users who commented.

Link to the json [feed](https://graph.facebook.com/126893747338271/feed?fields=id%2Cfull_picture%2Cmessage%2Cstory%2Ccreated_time%2Clink%2Ccomments&access_token=1786614831603417|DWqP48Lm77hBqCQDWVsq4lFixuo)

### CODE REVIEW

**Review by nikit19 — COMMENTED** [review]
But now comments from FOSSASIA won't be visible right ?

**Comment by dreadpool2** [comment]
@nikit19 Actually after the changes the comments are visible but the name of the commenter isn't. Comments written by FOSSASIA have both comment and commenter name (that is FOSSASIA ) visibility
32387159_1717279224974044_5092816647291404288_n

**Comment by dreadpool2** [comment]
@iamareebjamal Please review

File: android/app/src/main/java/org/fossasia/openevent/core/feed/facebook/CommentsListAdapter.java [hunk 1]
```diff
@@ -74,9 +74,11 @@ public void onBindViewHolder(RecyclerViewHolder holder, int position) {
             holder.comment.setVisibility(View.GONE);
         }
 
-        if (!TextUtils.isEmpty(commentItem.getFrom().getName())) {
-            holder.commenter.setText(commentItem.getFrom().getName());
-            holder.commenter.setVisibility(View.VISIBLE);
+        if (commentItem.getFrom() != null) {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/core/feed/facebook/CommentsListAdapter.java]: Extract comment.getFrom() above

Comment (dreadpool2) [reply android/app/src/main/java/org/fossasia/openevent/core/feed/facebook/CommentsListAdapter.java]: @iamareebjamal PR updated

File: android/app/src/main/java/org/fossasia/openevent/core/feed/facebook/CommentsListAdapter.java [hunk 2]
```diff
@@ -55,28 +55,30 @@ public RecyclerViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
...
             holder.comment.setVisibility(View.VISIBLE);
         } else {
             // status is empty, remove from view
             holder.comment.setVisibility(View.GONE);
         }
 
-        if (!TextUtils.isEmpty(commentItem.getFrom().getName())) {
-            holder.commenter.setText(commentItem.getFrom().getName());
-            holder.commenter.setVisibility(View.VISIBLE);
+        if (comment.getFrom() != null) {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/core/feed/facebook/CommentsListAdapter.java]: Still here. I meant

```Java
From from = comment.getFrom();
if (from != null) {
    String name = from.getName();
    if(!TextUtils.isEmpty(name)) {
           holder.commenter.setText(name);
           holder.commenter.setVisibility(View.VISIBLE);
       }
}
```

Also, why not combine the two `if`s by AND?

Also, revert other changes

Comment (dreadpool2) [reply android/app/src/main/java/org/fossasia/openevent/core/feed/facebook/CommentsListAdapter.java]: Ohh! I will do that :+1:

Comment (dreadpool2) [reply android/app/src/main/java/org/fossasia/openevent/core/feed/facebook/CommentsListAdapter.java]: PR updated

**Review by iamareebjamal — APPROVED** [review]

**Comment by nikit19** [comment]
The app is still crashing on opening the comments. I will try to solve it

**Pull request merged**